new resource: azurerm_firewall_policy_rule_collection_group (#8603)
terraform-providers/terraform-provider-azurerm: website/docs/r/firewall_policy_rule_collection_group.html.markdown

    rule {
      name                  = "network_rule_collection1_rule1"
      protocols             = ["TCP", "UDP"]
      source_addresses      = ["10.0.0.1"]
      destination_addresses = ["192.168.1.1", "192.168.1.2"]
      destination_ports     = ["80", "1000-2000"]
    }
  }

  nat_rule_collection {
    name     = "nat_rule_collection1"
    priority = 300
    action   = "Dnat"
    rule {
      name                = "nat_rule_collection1_rule1"
      protocols           = ["TCP", "UDP"]
      source_addresses    = ["10.0.0.1", "10.0.0.2"]
      destination_address = "192.168.1.1"
      destination_ports   = ["80", "1000-2000"]
      translated_address  = "192.168.0.1"
      translated_port     = "8080"
    }
  }
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name which should be used for this Firewall Policy Rule Collection Group. Changing this forces a new Firewall Policy Rule Collection Group to be created.

* `firewall_policy_id` - (Required) The ID of the Firewall Policy where the Firewall Policy Rule Collection Group should exist. Changing this forces a new Firewall Policy Rule Collection Group to be created.

> * `priority` - (Required) The priority of the Firewall Policy Rule Collection Group. The range is 100-65000.

Reviewer: ```suggestion
* `priority` - (Required) The priority of the Firewall Policy Rule Collection Group. The range is `100`-`65000`.
```

---

* `application_rule_collection` - (Optional) One or more `application_rule_collection` blocks as defined below.

* `nat_rule_collection` - (Optional) One or more `nat_rule_collection` blocks as defined below.

* `network_rule_collection` - (Optional) One or more `network_rule_collection` blocks as defined below.

---

A `application_rule_collection` block supports the following:

* `name` - (Required) The name which should be used for this application rule collection.

> * `action` - (Required) The action type of the application rule collection. Possible values are `Allow` and `Deny`.

Reviewer: ```suggestion
* `action` - (Required) The action to take for application rules in this collection. Possible values are `Allow` and `Deny`.
```

> * `priority` - (Required) The priority of the application rule collection. The range is 100-65000.

Reviewer: ```suggestion
* `priority` - (Required) The priority of this application rule collection. The range is `100`-`65000`.
```

* `rule` - (Required) One or more `rule` (network rule) blocks as defined above.

---

A `nat_rule_collection` block supports the following:

* `name` - (Required) The name which should be used for this nat rule collection.

* `action` - (Required) The action type of the nat rule collection. Currently, the only possible value is `Dnat`.

* `priority` - (Required) The priority of the nat rule collection. The range is 100-65000.

* `rule` - (Required) A `rule` (nat rule) block as defined above.

---

A `rule` (application rule) block supports the following:

* `name` - (Required) The name which should be used for this rule.

* `protocols` - (Required) One or more `protocols` blocks as defined below.

* `source_addresses` - (Optional) Specifies a list of source ip addresses.

* `source_ip_groups` - (Optional) Specifies a list of source ip groups.

* `destination_fqdns` - (Optional) Specifies a list of destination FQDNs.

* `destination_fqdn_tags` - (Optional) Specifies a list of destination FQDN tags.

---

A `rule` (network rule) block supports the following:

* `name` - (Required) The name which should be used for this rule.

> * `protocols` - (Required) Specifies a list of network protocols. Possible values are `Any`, `TCP`, `UDP`, `ICMP`.

Reviewer: ```suggestion
* `protocols` - (Required) Specifies a list of network protocols this rule applies to. Possible values are `Any`, `TCP`, `UDP`, `ICMP`.
```

> * `source_addresses` - (Optional) Specifies a list of source ip addresses.

Reviewer: IP in IP address should be capitalized
```suggestion
* `source_addresses` - (Optional) Specifies a list of source IP addresses.
```
